### ISSUE TITLE
[Question] Interoperation with a mod like Lockdown

### ISSUE THREAD

**Issue by rockobonaparte** [McJty/RFTools#408]
I was hoping to use RFTool's creative dimension builder to construct a tutorial world dimension for Baby's First Space Race, my tutorial mod pack. Another mod named "Lockdown" generally can provide a pregenerated world--like a skyblock--so I thought I'd use that to start the player in this tutorial dimension. I then wanted to transport the player to the Overworld, but have that be undefined so their own generator settings take over.

The built dimension does not survive the copy and restoration. Lockdown works by copying the save to a templated folder, and then it restores from that. The built dimension--in particular--gets destroyed. One of the important bits here is that I am creating the dimension in my original overworld, and then deleting that DIM folder when moving it to the template. So all the dimlets, the builder, and the supplier are there. Will that trigger a dimension delete?

If I did an /rftdim list, I can see Minecraft's basic dimensions, as well as The Last Millennium. The extra dimension I built is gone.

Can you foresee any ways RFTools may interfere in this? I am just asking you first because I have been told you are responsive. I am wondering if I have to somehow move all the infrastructure into the tutorial dimension in creative and encase it in bedrock or something.

PS: To handle teleporting the player to the overworld, I had to write my own little mod since there wasn't quite anything else out there to send a player to an as-yet-undefined overworld spawn point in 1.7.10.

**Comment by McJty** [comment]
Hmm, I have no experience with LockDown but I do know that the old FTB pathfinder modpack used this too and that modpack was completely based on RFTools dimensions. So I'm sure it should work. At least I can't think of a reason that it wouldn't.

But note that you do need a dimension builder with the dimension tab in it for this to work. RFTools cannot keep the dimension active without that.

**Comment by rockobonaparte** [comment]
I did build a dimension, so I should have a tab involved. Well, I did at least. I had a followup question here. I deleted the overworld dimension when copying the save to the template for Lockdown; I want the player's instance to generate its own overworld. This has a side effect of deleting all the infrastructure that was running the dimension. Would RFtools detect this as, say, an orphan dimension and remove it?

**Comment by McJty** [comment]
Well the dimension would get no power in that case (even a dimension build with a creative dimension builder still needs power). My recommendation would be to make a private/hidden void dimension where you put all your infrastructure (dimension builder + spotloader and such) to keep the other dimensions running. i.e. a dimension where the player cannot enter

**Comment by rockobonaparte** [comment]
Right now I have actually managed to dump it into the same dimension due to creative mode shenanigans and the /rftdim command. It seems to work fine, but I have not pushed it for too long.

My problem actually appears to be with Lockdown. Since I wanted to allow regular overworld generation, I tried to set "Disable Regular World Creation" to false. Apparently this is just a master switch to Lockdown, and it disables it completely.

I found setting that to true had me starting in the tutorial dimension perfectly fine. I did then try to delete the Overworld and go to the new overworld spawn, but it's frozen on the settings I used when I originally made the save that Lockdown uses as a template.

Anyways, it's an issue with Lockdown, or I am rather taking it up as a feature request there:
https://github.com/Adubbz/Lockdown/issues/7

So I'll close it here.

For reference to future modpack developers: Lockdown works fine in conjunction with RFTools so long as you want to freeze the overworld you used too. Maybe upon following the link to that issue, you will find some progress changing that if that somehow bothers you. If it's slow-going on tricky, I guess I'll check out Lockdown's source and attempt to implement the feature myself.
